src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -488,11 +498,11 @@ std::string JSIdent(const GeneratorOptions& options,
   } else {
     if (field->type() == FieldDescriptor::TYPE_GROUP) {
       result = is_upper_camel
-                   ? ToUpperCamel(ParseUpperCamel(field->message_type()->name()))
-                   : ToLowerCamel(ParseUpperCamel(field->message_type()->name()));
+                   ? ToUpperCamel(field->message_type()->name())
+                   : ToLowerCamel(field->message_type()->name());
     } else {
-      result = is_upper_camel ? ToUpperCamel(ParseLowerUnderscore(field->name()))
-                              : ToLowerCamel(ParseLowerUnderscore(field->name()));
+      result = is_upper_camel ? ToUpperCamel(field->name())
+                              : ToLowerCamel(field->name());
```

Comment:
before when we pass in `firstName`, the `ParseLowerUnderscore` will generate `['firstName']` instead of `['first', 'name']`